### PR TITLE
[impl] Add ascii codepage

### PR DESCRIPTION
**Pull request by tmoers**

### CODE REVIEW

File: src/byte_mapping.rs [hunk 1]
```diff
@@ -1,5 +1,30 @@
 const NIL: char = '';
 
+/// The mapping for [ASCII](https://en.wikipedia.org/wiki/ASCII)
+///
+/// This mapping uses the standard ascii character set which is composed of a
+/// 7-bit code (or 128 characters). The first 32 characters and the last one
+/// are non-printable control characters.
+pub const CODEPAGE_ASCII: &'static [char]  = &[
+//   0     1     2     3     4     5     6     7     8     9     A     B     C     D     E     F
+    NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL,  NIL, // 0
```
Comment (flier) [inline src/byte_mapping.rs]: Can we use '.' as placeholder instead of space, which could help us count the missed characters.

Comment (tmoers) [inline src/byte_mapping.rs]: Sure, I could change that.. But the current NIL character shouldn't be whitespace.. This is what I have:
image

So this is not what you see?

Comment (tmoers) [inline src/byte_mapping.rs]: Ah no, I see.. I get `?` on the windows console, and `` on bash. I suppose it depends on the font?

Comment (flier) [inline src/byte_mapping.rs]: That's why I suggest to use '.', I got a lot of messy code when run the code in different platform with different console/term, if you real need a special symbol, we may use [rust-encoding](https://github.com/lifthrasiir/rust-encoding) for compatibility

Comment (flier) [inline src/byte_mapping.rs]: Besides, on my macbook, I use [powerline-fonts](https://github.com/powerline/fonts) which also got problems :(

**Comment by tmoers** [comment]
Ok, good to know. I'll take a look at rust-encoding when I get home this
evening. If it's just too much of a hassle I'll change NIL to use `.`.
Thanks for pointing this out!

**Comment by flier** [comment]
👍

**Comment by tmoers** [comment]
I pushed a new version and changed the fallback character to the default
unicode fallback character. Could you try this?
If this still doesn't work, maybe it's just because your
console/font/whatever isn't capable of showing unicode characters?
Then we may need an option to specify the character or something..

**Comment by flier** [comment]
Yes, it works, but it still hard to read, because most of time I need ignore those

54ab37a7-94f3-4734-b21b-41c16e29cbc8

Would you mind to add the replacement charater to the config?

```rust
HexViewBuilder::new(buf)
    .codepage(CODEPAGE_ASCII)
    .address_offset(off)
    .placeholder('.') 
    .row_width(16)
    .finish());
```
and we could get a more clear ascii view

16b73cfe-1e7f-4ac6-90be-a7f0daa55750

**Comment by tmoers** [comment]
This is just plain ugly. I made the replacement character configurable as
you requested. But I also changed the default
from std::char::REPLACEMENT_CHARACTER to `.`.

The new version is also on crates.io

**Comment by flier** [comment]
Great! Thanks very much!

**Pull request merged**